engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceModificationProcessor.java

```diff
@@ -145,7 +151,7 @@ private void executeGlobalVariableInstructions(
                     variableDocument));
   }
 
-  private void terminateElement(
+  private long terminateElement(
```

Comment:
🔧 The way I read this, the reason why this method returns the key of the flow scope is that it's useful. But semantically, it makes more sense for this method to return the element instance that was terminated. You could then still retrieve the flow scope key from it between lines 115 and 117, as well as on line 184.